openxr/Cargo.toml

```diff
@@ -32,5 +32,40 @@ ctrlc = "3.1.5"
 [target.'cfg(windows)'.dependencies]
 winapi = { version = "0.3", features = ["profileapi"] }
 
+[target.'cfg(target_os = "android")'.dependencies]
+ndk-glue = "0.3"
+
 [package.metadata.docs.rs]
 features = ["linked", "loaded", "mint"]
+
+[[example]]
+name = "vulkan"
+
+[[example]]
+name = "vulkan-android"
+path = "examples/vulkan.rs"
+crate-type = ["cdylib"]
```

Comment:
Looks like this is causing lint issues on non-android. Can we guard the example to android-only somehow? Failing that, we should move most of the code into a common module, and add suitable conditional lint `allow`s in the vulkan-android-specific source file, perhaps combined with a `panic` with a suitable message if ran on non-android.

Reply:
I'll look into it.

Reply:
The solution was to add also the "bin" crate-type. I've seen doing this elsewhere but I wasn't aware it fixes the lints.

Reply:
Huh. Does this result a useless binary being produced as well? Is the android build process unaffected?

Reply:
I just tried, the binary gets created in the target folder, but it doesn't get used. it doesn't seem to affect compilation times.

Reply:
I added a commit to use `#[allow(unused)]` instead. This is something that can be integrated in ndk_glue.

Reply:
I found a more elegant solution, make the main `pub`.